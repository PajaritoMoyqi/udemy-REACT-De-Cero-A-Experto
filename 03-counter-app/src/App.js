// return html to index.js(padre)

import React from 'react'; // to use jsx
import propTypes from 'prop-types';

const App = ({ saludo, subtitulo }) => {
  if(!saludo){
    throw new Error('El saludo es necesario');
  }

  return (
    <>
      <h1> { saludo } </h1>
      <p> { subtitulo } </p>
    </>
  );
}

// need property from index.js

App.propTypes = {
  saludo: propTypes.string.isRequired
}  // variable condition
App.defaultProps = {
  subtitulo: 'Mi Primera Aplicacion'
}  // variable default. no need any import. why?

export default App; // export App.js