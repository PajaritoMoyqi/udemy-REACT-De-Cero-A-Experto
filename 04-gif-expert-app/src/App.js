import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const App = ({defaultCat = []}) => {

  // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
  const [categories, setCategories] = useState(defaultCat);

  const handleAdd = () => {
    const name = '충사';

    setCategories(cats => [...cats, name]);
  }
  
  return  (
    <>
      <h2>Gif-Expert-App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        { categories.map((item)=>
          <GifGrid key={item} category={item}
        />) }
      </ol>
    </>
  );

} // key는 무조건 유니크해야 함.

export default App;