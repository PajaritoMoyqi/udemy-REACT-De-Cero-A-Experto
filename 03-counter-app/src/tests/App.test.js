import App from '../App'
import React from 'react';
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import "../setupTests"; // 내거에서는 왜 이 파일이 작동하지 않는지 모르겠다.
import { shallow } from 'enzyme';

describe('App test', ()=>{
  
  // test('show title', ()=>{

  //   const title = 'Hola Mundo';

  //   const {getByText} = render( <App saludo={title} /> )

  //   expect(getByText(title)).toBeInTheDocument();

  // })

  test('show title with enzyme', () => {

    const saludo = 'Hola Mundo'
    const wrapper = shallow( <App saludo={saludo} />);

    expect( wrapper ).toMatchSnapshot();

  })
  
  test('show sub with enzyme', () => {

    const saludo = 'Hola Mundo'
    const sub = 'Mi Primera Aplicacion'
    const wrapper = shallow( <App saludo={saludo} subtitulo={sub} />);
    const parrafo = wrapper.find('p').text().trim();

    expect( parrafo ).toBe(sub);

  })

})