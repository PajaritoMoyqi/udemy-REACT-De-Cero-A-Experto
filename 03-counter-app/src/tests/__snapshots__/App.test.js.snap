// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test show title with enzyme 1`] = `
<Fragment>
  <h1>
     
    Hola Mundo
     
  </h1>
  <p>
     
    Mi Primera Aplicacion
     
  </p>
</Fragment>
`;
